worker/uniter/upgrade126.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package uniter
+
+import (
+	"github.com/juju/names"
+	"gopkg.in/juju/charm.v5/hooks"
+
+	"github.com/juju/juju/worker/uniter/operation"
+)
+
+func AddInstalledToUniterState(tag names.UnitTag, dataDir string) error {
+	logger.Tracef("entering upgrade step AddInstalledToUniterState")
+	defer logger.Tracef("leaving upgrade step AddInstalledToUniterState")
+
+	opsFile := getUniterStateFile(dataDir, tag)
+	state, err := readUnsafe(opsFile)
+	switch err {
+	case nil:
+		return addInstalled(opsFile, state)
+	case operation.ErrNoStateFile:
+		logger.Warningf("no uniter state file found for unit %s, skipping uniter upgrade step", tag)
+		return nil
+	default:
+		return err
+	}
+}
+
+func addInstalled(opsFile string, state *operation.State) error {
+	statefile := operation.NewStateFile(opsFile)
+	if state.Kind == operation.Install || state.Kind == operation.RunHook && state.Hook.Kind == hooks.Install {
+		state.Installed = false
```

Comment:
do we really need this line?
why not just write of the file if Installed is false and it should be true?